core-android/src/main/java/org/hisp/dhis/client/sdk/android/program/ProgramApiClientImpl.java

```diff
@@ -110,6 +115,39 @@ public String getDescendantProperties() {
             }
         };
 
-        return getCollection(apiResource, fields, lastUpdated, uids);
+        List<Program> programs = getCollection(apiResource, fields, lastUpdated, uids);
+
+        for (Program program : programs) {
+            if (program.getProgramStages() != null && !program.getProgramStages().isEmpty()) {
+                for (ProgramStage programStage : program.getProgramStages()) {
+                    if (programStage.getProgramStageSections() != null && !programStage.getProgramStageSections().isEmpty()) {
+                        for (ProgramStageSection programStageSection : programStage.getProgramStageSections()) {
+                            if (programStageSection.getProgramStageDataElements() != null && !programStageSection.getProgramStageDataElements().isEmpty()) {
+                                for (int i = 0; i < programStageSection.getProgramStageDataElements().size(); i++) {
+                                    ProgramStageDataElement programStageDataElement = programStageSection.getProgramStageDataElements().get(i);
+                                    programStageDataElement.setSortOrderWithinProgramStageSection(i);
+                                }
+                            }
+
+                            if (programStage.getProgramStageDataElements() != null && !programStage.getProgramStageDataElements().isEmpty()) {
+                                for (ProgramStageDataElement programStageDataElement : programStage.getProgramStageDataElements()) {
+                                    if (programStageDataElement.getDataElement() != null && programStageDataElement.getDataElement().getOptionSet() != null) {
+                                        OptionSet optionSet = programStageDataElement.getDataElement().getOptionSet();
+
+                                        if (optionSet.getOptions() != null) {
+                                            for (int i = 0; i < optionSet.getOptions().size(); i++) {
+                                                Option option = optionSet.getOptions().get(i);
+                                                option.setSortOrder(i);
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return programs;
```

Comment:
You have 10 levels nested here. I'm sure you can avoid this by using functions. Please find a beautiful way to transform this code.